src/org/zaproxy/zap/extension/ascanrulesBeta/UsernameEnumeration.java

```diff
@@ -545,6 +554,17 @@ public String longestCommonSubsequence (String a, String b) {
 		return hirschberg.getLCS(a, b);
 	}
 
+	private boolean shouldContinue(List<Context> contextList) {
+		Iterator<Context> contextIterator = contextList.iterator();
+		boolean hasAuth = false;
+		while (contextIterator.hasNext()) {
+			if(contextIterator.next().getAuthenticationMethod().getType().getName().equals("Form-based Authentication")) {
+				hasAuth = true;
```

Comment:
This could just return true (or break the loop), it does not need to check other contexts.